tests/Unit/DataStructures/Test_Mesh.cpp

```diff
@@ -223,6 +225,10 @@ SPECTRE_TEST_CASE("Unit.DataStructures.Mesh", "[DataStructures][Unit]") {
   }
 }
 
+SPECTRE_TEST_CASE("Unit.Domain.MeshTag", "[Unit][Domain]") {
+  CHECK(Tags::Mesh<3>::name() == "Mesh");
```

Comment:
Please put this in the `Mesh` test so we don't have to spend time waiting for an additional test to start. Tests tend to have a fairly high overhead on some HPC systems.